mars/services/task/execution/api.py

```diff
@@ -14,12 +14,124 @@
 
 from abc import ABC, abstractmethod
 from dataclasses import dataclass
-from typing import List, Dict, Any, Type
+from typing import List, Dict, Any, Type, Union
 
 from ....core import ChunkGraph, Chunk, TileContext
 from ....resource import Resource
 from ....typing import BandType
+from ....utils import merge_dict
 from ...subtask import SubtaskGraph, SubtaskResult
+from .core import _CommonPrivateConfigMixin
+
+
+class ExecutionConfig(_CommonPrivateConfigMixin):
+    """
+    The config for execution backends.
+
+    This class should ONLY provide the APIs for the parts other than
+    just the execution. Each backend may have a different implementation
+    of the API.
+
+    If there are common logic shared by all the backends, but they are
+    only used inside the execution. The `_CommonPrivateConfigMixin` is
+    a good place for them.
+
+    If some configuration is for a specific backend. They should be in
+    the backend config. e.g. `get_mars_special_config()` should be in
+    the `MarsExecutionConfig`.
+    """
+
+    name = None
+
+    def __init__(self, execution_config: Dict):
+        """
+        An example of execution_config:
+        {
+            "backend": "mars",
+            "mars": {
+                "n_worker": 1,
+                "n_cpu": 2,
+                ...
+            },
+        }
+        """
+        self._execution_config = execution_config
+
+    def merge_from(self, execution_config: "ExecutionConfig") -> "ExecutionConfig":
+        assert isinstance(execution_config, ExecutionConfig)
+        assert self.backend == execution_config.backend
+        merge_dict(
+            self._execution_config,
+            execution_config.get_execution_config(),
+        )
+        return self
+
+    @property
+    def backend(self) -> str:
+        """The backend from config."""
+        return self._execution_config["backend"]
+
+    def get_execution_config(self) -> Dict:
+        """Get the execution config dict."""
+        return self._execution_config
+
+    @abstractmethod
+    def get_deploy_band_resources(self) -> List[Dict[str, Resource]]:
+        """Get the band resources for deployment."""
+
+    @classmethod
+    def from_config(cls, config: Dict, backend: str = None) -> "ExecutionConfig":
```

Comment:
This passes the whole mars config, do we really need this? Seems `from_execution_config` is enough.

Reply:
This is for merging the execution config part of the mars config: https://github.com/mars-project/mars/pull/2968/files#diff-1b98ea665f30f15485deb7982cdb0fc5e9127cb2225b9197811135ecbaefee73R156

With out this API, we have to access the execution config from dict by some keys, it is not maintainable.